packages/algorand-js/src/interpreter/opcode-list.ts

```diff
@@ -151,3 +152,40 @@ export class Intc extends Op {
     stack.push(intc);
   }
 }
+
+// pop a value from the stack and store to scratch space
+export class Store extends Op {
+  readonly index: number;
+  readonly interpreter: Interpreter;
+
+  constructor (index: number, interpreter: Interpreter) { // eslint-disable-line sonarjs/no-identical-functions
+    super();
+    this.index = index;
+    this.interpreter = interpreter;
+  }
+
+  execute (stack: TEALStack): void {
+    this.checkIndexBound(this.index, this.interpreter.scratch);
+    this.assertStackLen(stack, 1);
+    const top = stack.pop() as StackElem;
```

Comment:
how possible `stack.pop()` doesn't return `StackElem`?

Reply:
That's because the `pop()` definition of stack returns `T | undefined`. 
![image](https://user-images.githubusercontent.com/33264364/100375428-c9151700-3033-11eb-8ab8-fa63c5d3cbbf.png)
I am changing it to `T`.

Reply:
1. You have typo: UnderFlow -> underflow
2. I think underflow is a wrong word here. Use `Error("pop from empty stack")`

Reply:
Ok. updated.